zerver/views/message_summary.py

```diff
@@ -0,0 +1,131 @@
+import json
+from typing import Any
+
+import litellm
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+from django.utils.timezone import now as timezone_now
+from pydantic import Json
+
+from analytics.lib.counts import COUNT_STATS, do_increment_logging_stat
+from zerver.lib.narrow import NarrowParameter
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import typed_endpoint
+from zerver.models import UserProfile
+from zerver.views.message_fetch import get_messages_backend
+
+# Maximum number of messages that can be summarized in a single request.
+MAX_MESSAGES_SUMMARIZED = 100
+OUTPUT_COST_PER_TOKEN = 5
+INPUT_COST_PER_TOKEN = 1
+
+
+def format_conversation(zulip_messages: list[dict[str, Any]]) -> str:
```

Comment:
`format_zulip_messages_for_model` might be clearer.

Reply:
Renamed.